docs/api/js/utils.md

```diff
@@ -1,68 +1,563 @@
+export const IERC20 = new utils.Interface(require('../../abi/IERC20.json').abi);
+```
+
+#### IERC1271
+
+```ts
+export const IERC1271 = new utils.Interface(require('../../abi/IERC1271.json').abi);
 ```
 
-zkSync also allow a more friendly zero address alias in our SDK and API:
+#### Contract deployer
+
+Used for deploying smart contracts.
+
+```ts
+export const CONTRACT_DEPLOYER = new utils.Interface(require('../../abi/ContractDeployer.json').abi);
+```
+
+#### L1 messenger
+
+Used for sending messages from zkSync Era to Ethereum.
+
+```ts
+export const L1_MESSENGER = new utils.Interface(require('../../abi/IL1Messenger.json').abi);
+```
+
+### Useful addresses
+
+#### ETH token layer 1
 
 ```typescript
-export const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const ETH_ADDRESS = '0x0000000000000000000000000000000000000000';
 ```
 
-## ABI of zkSync smart contract
+#### ETH token alias on ZkSync Era
 
 ```typescript
-export const ZKSYNC_MAIN_ABI = new utils.Interface(require("../abi/IZkSync.json"));
+export const L2_ETH_TOKEN_ADDRESS = '0x000000000000000000000000000000000000800a';
 ```
 
-## IERC20 interface
+#### Bootloader
 
-Convenient when interacting with native tokens on zkSync.
+```ts
+export const BOOTLOADER_FORMAL_ADDRESS = '0x0000000000000000000000000000000000008001';
+```
 
-```typescript
-export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
+#### Contract deployer
+
+```ts
+export const CONTRACT_DEPLOYER_ADDRESS = '0x0000000000000000000000000000000000008006';
+```
+
+#### L1 messenger
+
+```ts
+export const L1_MESSENGER_ADDRESS = '0x0000000000000000000000000000000000008008';
 ```
 
-## Encoding paymaster params
+### Gas
+
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
+:::
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
 ```
 
-The definition of the `PaymasterInput` can be found [here](./types.md).
+#### `REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`
 
-## Useful gas constants
-Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
-Later on it will be possible to query the current recommended limit.
+Any amount is possible for L1->L2 transactions because the cost per gas is adjusted.
 
-```typescript
-const GAS_PER_PUBDATA_BYTE = 17;
+```ts
+export const REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT = 800;
+```
```

Comment:
for some reasons there is no mentrion about these functions:
```
export function isETH(token: Address) {
    return token.toLowerCase() == ETH_ADDRESS || token.toLowerCase() == L2_ETH_TOKEN_ADDRESS;
}

export function sleep(millis: number) {
    return new Promise((resolve) => setTimeout(resolve, millis));
}

export function layer1TxDefaults() {
    return {
        queueType: PriorityQueueType.Deque,
        opTree: PriorityOpTree.Full
    };
}
```

Reply:
i will add these ->  i thought they were maybe uninteresting to users ...